src/dbnode/client/session.go

```diff
@@ -1547,20 +1547,20 @@ func (s *session) fetchTaggedAttempt(
 	// once https://github.com/m3db/m3ninx/issues/42 lands. Including transferring ownership
 	// of the Clone()'d value to the `fetchState`.
 	const fetchData = true
-	if opts.InstanceMultiple > 0 {
-		topo := s.state.topoMap
-		iPerReplica := len(topo.Hosts()) / topo.Replicas()
-		iSeriesLimit := int(float32(opts.SeriesLimit)*opts.InstanceMultiple) / iPerReplica
-		if iSeriesLimit < opts.SeriesLimit {
-			opts.SeriesLimit = iSeriesLimit
-		}
-	}
 	req, err := convert.ToRPCFetchTaggedRequest(nsClone, q, opts, fetchData)
 	if err != nil {
 		s.state.RUnlock()
 		nsClone.Finalize()
 		return nil, FetchResponseMetadata{}, xerrors.NewNonRetryableError(err)
 	}
+	if opts.InstanceMultiple > 0 {
+		topo := s.state.topoMap
+		iPerReplica := int64(len(topo.Hosts()) / topo.Replicas())
+		iSeriesLimit := int64(float32(opts.SeriesLimit)*opts.InstanceMultiple) / iPerReplica
+		if req.SeriesLimit != nil && iSeriesLimit < *req.SeriesLimit {
```

Comment:
what would `req.SeriesLimit` have previously been set to? `opts.SeriesLimit`?

Reply:
Yes